internal/controller/person_test.go

```diff
@@ -1,15 +1,62 @@
 package controller
 
-import "testing"
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"testGoApi/internal/middlewares"
+	"testGoApi/internal/models"
+	"testGoApi/internal/test_helpers"
+)
+
+func setPersonContext() context.Context {
+	return middlewares.SetPersonCtx(context.Background(), &models.Person{})
+}
 
 func TestPersonController_HandleGetAllPerson(t *testing.T) {
+	controller := NewPersonController(&test_helpers.MockPersonService{})
+
+	req := test_helpers.NewRequest(t, http.MethodGet, "/person", nil)
+
+	rr := test_helpers.ExecuteRequest(req, controller.HandleGetAllPerson, nil)
 
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("Handler returned wrong status code. Expected: %d. Got: %d.", http.StatusOK, status)
+	}
 }
 
 func TestPersonController_HandleGetPerson(t *testing.T) {
+	controller := NewPersonController(&test_helpers.MockPersonService{})
```

Comment:
let's follow New -> pattern since we are not using the struct locally but rather calling it from another file , same goes to the Movie as well.

Reply:
Didn't get your point here
Do you mean to have func New for each struct?

Reply:
Yes for the mock ones since we are using it outside of where it was created.